src/handlers/expiry/providers/DateProvider.ts

```diff
@@ -0,0 +1,139 @@
+   * @param input
+   */
+  public static isValidDate(
+    input: string | Date | number | undefined
+  ): boolean {
+    return (
+      input !== undefined &&
+      moment(input).isValid() &&
+      moment(input).isAfter(new Date(0))
+    );
+  }
+  /**
+   * To compare whether input date is between two months of compare date. Inlcusivty parameter represents start and end of month.
+   * @param inputDate the input date.
+   * @param compareDate the date to compare with.
+   * @param inclusivity []= start and end included, ()= start and end excluded, [)= start included end excluded, (]= start excluded and end included.
+   */
+  public static IsBetweenTwoMonths(
+    inputDate: Date,
+    compareDate: Date,
+    inclusivity: "[]" | "()" | "[)" | "(]" | undefined
+  ): boolean {
+    console.log(`compareDate+2 months: ${moment(compareDate).add(2, "months").toISOString()}`);
+    return moment(inputDate).isBetween(
+      moment(compareDate),
+      moment(compareDate).add(2, "months"),
+      "days",
+      inclusivity
+    );
+  }
+
+  public static IsAfterCompareDate(
+    inputDate: Date,
+    compareDate: Date
+  ): boolean {
+    return moment(inputDate).isAfter(compareDate, "days");
+  }
+
+  public static IsBeforeTwoMonths(
+    inputDate: Date,
+    compareDate: Date
+  ): boolean {
+    return moment(inputDate).isSameOrBefore(moment(compareDate).subtract(2, "months").startOf("day"), "days");
+  }
+
+  public static GetEpoc(): Date {
+    return new Date(0);
+  }
+
+
+
+  public static addOneYear(inputDate: Date | string): Date {
+    return moment(inputDate)
+      .add(1, "years")
+      .toDate();
+  }
+
+  public static addOneYearStartOfMonth(inputDate: Date | string): Date {
+    return moment(inputDate).add(1, "years").startOf("month").startOf("day").toDate();
+  }
+
+  public static addOneYearEndOfMonth(inputDate: Date | string): Date {
+    return moment(inputDate).add(1, "year").endOf("month").startOf("day").toDate();
+  }
+
+  public static getEndOfMonth(inputDate: Date | string): Date {
+    return moment(inputDate).endOf("month").startOf("day").toDate();
+  }
+
+  public static addOneYearISOString(inputDate: Date | string): string {
+    return moment(inputDate)
+      .add(1, "years")
+      .toISOString();
+  }
+
+  public static addOneYearStartOfDayISOString(inputDate: Date | string): string {
+    return moment(inputDate)
+      .add(1, "year")
+      .startOf("day")
+      .toISOString();
+  }
+
+  public static addOneYearMinusOneDayISOString(inputDate: Date | string): string {
+    return moment(inputDate)
+      .add(1, "year")
+      .subtract(1, "day")
+      .startOf("day")
+      .toISOString();
+  }
+
+  public static getLastDayOfMonthInNextYearISOString(inputDate: Date | string): string {
+    return moment(inputDate)
+      .add(1, "year")
+      .endOf("month")
+      .startOf("day")
+      .toISOString();
+  }
+
+  public static getStartOfMonthInNextYearISOString(inputDate: Date | string): string {
```

Comment:
some methods that are not being used, can we remove them?

Reply:
removed